File: packages/domain/src/entities/log.rs
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct LogRecord {
    pub id: String,
    pub time_ns: u64,
    pub observed_time_ns: Option<u64>,
    pub severity_number: Option<i32>,
    pub severity_text: Option<String>,
    pub body_any: Option<Value>,
    pub event_name: Option<String>,
    pub service_name: String,
    pub resource_attributes: Value,
    pub resource_dropped_attributes_count: u32,
    pub resource_schema_url: Option<String>,
    pub scope_name: Option<String>,
    pub scope_version: Option<String>,
    pub scope_attributes: Value,
    pub scope_dropped_attributes_count: u32,
    pub scope_schema_url: Option<String>,
    pub attributes: Value,
    pub dropped_attributes_count: u32,
    pub flags: u32,
    pub trace_id: Option<String>,
    pub span_id: Option<String>,
}

File: apps/api/src/features/logs/types/log.ts
import type { IngestProviderName, Json } from "@shared/helpers"

export type LogRecord = {
  id: string
  timeNs: bigint
  observedTimeNs?: bigint
  severityNumber?: number
  severityText?: string
  bodyAny?: Json
  eventName?: string
  serviceName: string
  resourceAttributes: Json
  resourceDroppedAttributesCount: number
  resourceSchemaUrl?: string
  scopeName?: string
  scopeVersion?: string
  scopeAttributes: Json
  scopeDroppedAttributesCount: number
  scopeSchemaUrl?: string
  attributes: Json
  droppedAttributesCount: number
  flags: number
  traceId?: string
  spanId?: string
  ingestProvider?: IngestProviderName
}

export type LogDto = {
  id: string
  time: string
  severity_number: number | null
  severity_text: string | null
  body: Json
  service_name: string
  attributes: Json
  scope_name: string | null
  scope_version: string | null
  trace_id: string | null
  span_id: string | null
  provider: string
}

export type LogSortField = "date" | "service" | "severity"
export type LogSortOrder = "asc" | "desc"

export type LogListFilters = {
  limit: number
  offset: number
  sort: LogSortField
  order: LogSortOrder
  service?: string
  severity?: string
  message?: string
  traceId?: string
  sinceNs?: bigint
  untilNs?: bigint
  raw?: boolean
}

export type FacetValue = {
  value: string
  count: number
}

export type LogFacets = {
  services: FacetValue[]
  severities: FacetValue[]
}

export type LogFacetsDto = {
  services: FacetValue[]
  severities: FacetValue[]
}
